python/ray/serve/_private/replica.py

```diff
@@ -713,6 +718,26 @@ async def initialize(self, deployment_config: DeploymentConfig):
         except Exception:
             raise RuntimeError(traceback.format_exc()) from None
 
+    async def update_replica_rank(self, rank: int, world_size: int) -> bool:
```

Comment:
can this be merged into using `reconfigure()`? we call the actor's method `reconfigure` today when various deployment configurations change, e.g. `graceful_shutdown_period_s`, which will update the replica's `graceful_shutdown_period_s` variable but won't actually call the user's `reconfigure()` method. 

can we make it so that when `num_replicas` is changed or the individual rank is changed, it will also call the actor's `reconfigure` method (same code path as `graceful_shutdown_period_s`)?

code ptr: https://github.com/ray-project/ray/blob/04b636f039334e956459f268860862580cbe6286/python/ray/serve/_private/deployment_state.py#L591